### PR TITLE
Week 1

### PR DESCRIPTION
**Pull request by ashubly25**
completed the first assignment

### CODE REVIEW

File: week1/delete_bst.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    if key < node.key:
+        node.left = insert(node.left, key)
+    else:
+        node.right = insert(node.right, key)
+    return node
+def minValueNode( node):
+    current = node
+    while(current.left is not None):
+        current = current.left 
+    return current
```
Comment (amanchoudhary) [inline week1/delete_bst.py]: Give a blank line after each method so that code is readable

File: week1/delete_bst.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    current = node
+    while(current.left is not None):
+        current = current.left 
+    return current 
+ def deleteNode(root, key):
+ if root is None:
+    return root 
+    if key < root.key:
+        root.left = deleteNode(root.left, key)
+    elif(key > root.key):
```
Comment (amanchoudhary) [inline week1/delete_bst.py]: avoid too many nested ifs
return when a condition satisfies

File: week1/delete_trie.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+ 
+ 
+    def _deleteHelper(self,pNode,key,level,length):
+        if pNode:
+            if level == length:
+                if pNode.value:
+                    pNode.value = 0
+                return pNode.isItFreeNode()
+ 
+        else:
```
Comment (amanchoudhary) [inline week1/delete_trie.py:62]: looks like there is an indentation issue

File: week1/deletion-linkedlist.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ def __init__(self):
+ self.head = None
+ 
+def push(self, new_data):
+        new_node = Node(new_data)
+        new_node.next = self.head
+        self.head = new_node
+ 
+def deleteNode(self, key):
+
```
Comment (amanchoudhary) [inline week1/deletion-linkedlist.py:15]: indentation issue

**Review by amanchoudhary — APPROVED** [review]
Changes suggested are mostly round code readability

File: week1/delete_bst.py (hunk 2 above)
Comment (ashubly25) [reply week1/delete_bst.py]: okay

File: week1/delete_bst.py (hunk 1 above)
Comment (ashubly25) [reply week1/delete_bst.py]: done

**Pull request closed without merging**